src/olympia/users/models.py

```diff
@@ -192,9 +187,26 @@ def has_module_perms(self, app_label):
         return self.is_superuser
 
     def has_read_developer_agreement(self):
+        from olympia.zadmin.models import get_config
+
         if self.read_dev_agreement is None:
             return False
-        return self.read_dev_agreement > self.last_developer_agreement_change
+        try:
+            last_agreement_change_config = get_config(
+                'last_dev_agreement_change_date')
+            change_config_date = datetime.strptime(
+                last_agreement_change_config, '%Y-%m-%d %H:%M')
+            now = datetime.now()
+
+            # If the config date is in the future, instead check against the
+            # current date
+            return self.read_dev_agreement > min(now, change_config_date)
```

Comment:
If the date is in the future, checking against `now` is pointless, as it will *also* be after the date the developer agreed to the agreement, by a few milliseconds if they just did. Because of this, setting a date in the future will show the agreement on every upload.

I think that instead, to handle future dates you need to compare `change_config_date` with `now`, and if the former is higher, raise `ValueError` in order to use `datetime(2017, 9, 22, 17, 36)` below.